tests/shardddl1/run.sh

```diff
@@ -0,0 +1,480 @@
+    run_sql_source1 "create database ${shardddl1};"
+    check_log_contain_with_retry "CREATE DATABASE IF NOT EXISTS \`${shardddl1}\`" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+}
+
+function DM_015() {
+    run_case 015 "single-source-pessimistic" "init_table 111" "clean_table 111" ""
+}
+
+function DM_016_CASE() {
+    run_sql_source1 "drop database ${shardddl1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+    run_sql_source1 "create database if not exists ${shardddl1};"
+    check_log_contain_with_retry "CREATE DATABASE IF NOT EXISTS \`${shardddl1}\`" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+}
+
+function DM_016() {
+    run_case 016 "single-source-pessimistic" "init_table 111" "clean_table" ""
+}
+
+function DM_017_CASE() {
+    run_sql_source1 "drop table ${shardddl1}.${tb1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log  $WORK_DIR/worker2/log/dm-worker.log
+    run_sql_source1 "create table ${shardddl1}.${tb1}(id int);"
+    check_log_contain_with_retry "CREATE TABLE IF NOT EXISTS \`${shardddl1}\`.\`${tb1}\` (\`id\` INT)" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log
+}
+
+function DM_017() {
+    run_case 017 "single-source-pessimistic" "init_table 111" "clean_table" ""
+}
+
+function DM_018_CASE() {
+    run_sql_source1 "drop table ${shardddl1}.${tb1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log 
+    run_sql_source1 "create table if not exists ${shardddl1}.${tb1}(id int);"
+    check_log_contain_with_retry "CREATE TABLE IF NOT EXISTS \`${shardddl1}\`.\`${tb1}\` (\`id\` INT)" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log
+}
+
+function DM_018() {
+    run_case 018 "single-source-pessimistic" "init_table 111" "clean_table" ""
+}
+
+function DM_019_CASE() {
+    run_sql_source1 "truncate table ${shardddl1}.${tb1};"
+    check_log_contain_with_retry "skip event, need handled ddls is empty" $WORK_DIR/worker1/log/dm-worker.log $WORK_DIR/worker2/log/dm-worker.log
+}
+
+function DM_019() {
+    run_case 019 "single-source-pessimistic" "init_table 111" "clean_table" ""
+}
+
+function DM_020_CASE() {
+    run_sql_source1 "rename table ${shardddl1}.${tb1} to ${shardddl1}.${tb2};"
+    run_sql_source1 "insert into ${shardddl1}.${tb2} values (1);"
+    run_sql_tidb_with_retry "show tables from ${shardddl1};" "${tb2}"
+    run_sql_tidb_with_retry "select count(1) from ${shardddl1}.${tb2};" "count(1): 1"
+}
+
+function DM_020() {
+    run_case 020 "single-source-no-routes" "init_table 111" "clean_table;run_sql_tidb \"drop database ${shardddl1};\"" ""
+}
+
+function DM_021_CASE() {
+    # same as "rename ${shardddl}.${tb} to ${shardddl}.${tb};"
```

Comment:
what is the comment means?

Reply:
It help us to understand the case. `rename tb1 to tb2` will be `tranfer to rename tb to tb`